mechanisms.py

```diff
@@ -122,7 +122,7 @@ def voltage_profile_setup(self, second_reduction_potential: float | None = None)
         for step in range(self.n_max):
             potential.append(theta)
             # Using <= here for exact midpoint
-            if step <= self.n_max // 2:
+            if step <= (self.n_max // 2) - 2:
```

Comment:
Why `-2`?

Reply:
The Oldham method takes inputs defining the potential excursion e.g., `start_potential`, `switch_potential`, `step_size`. It then calculates current for all steps in the potential sweep inclusive of the next step AFTER `start_potential`, to `switch_potential` and then back to `start_potential`. You have to really zoom in on the paper figs to see that this is the case (🙌 vector graphics).
So: `-1` for beginning on the _second_ step; `-1` for 0-based indexing. Results in `-2`.

This ensures that the scan reversal potential is `switch_potential`.